### PR TITLE
feat: Upgraded eks version 1.26 to 1.27

### PR DESCRIPTION
**Pull request by amanpruthi**

### CODE REVIEW

File: modules/app_eks/add-ons.tf [hunk 1]
```diff
@@ -37,8 +37,9 @@ resource "aws_eks_addon" "aws_efs_csi_driver" {
    ]
    cluster_name               = var.namespace
    addon_name                 = "aws-efs-csi-driver"
-   addon_version              = "v1.7.7-eksbuild.1"
+   addon_version              = "v2.0.3-eksbuild.1"
```
Comment (zacharyblasczyk) [inline modules/app_eks/add-ons.tf:40]: https://github.com/kubernetes-sigs/aws-efs-csi-driver?tab=readme-ov-file#kubernetes-version-compability-matrix  

v2.0.4 is out

File: modules/app_eks/add-ons.tf [hunk 2]
```diff
@@ -57,7 +59,7 @@ resource "aws_eks_addon" "coredns" {
   ]
   cluster_name                = var.namespace
   addon_name                  = "coredns"
-  addon_version               = "v1.9.3-eksbuild.11"
+  addon_version               = "v1.10.1-eksbuild.4"
```
Comment (zacharyblasczyk) [inline modules/app_eks/add-ons.tf:62]: https://github.com/aws/amazon-vpc-cni-k8s?tab=readme-ov-file#recommended-version
For all Kubernetes releases, we recommend installing the latest VPC CNI release

Comment (zacharyblasczyk) [reply modules/app_eks/add-ons.tf:62]: I think we need v1.12.5+ (is recommended) for 1.27

**Review by zacharyblasczyk — APPROVED** [review]

File: modules/app_eks/add-ons.tf (hunk 2 above)
Comment (zacharyblasczyk) [reply modules/app_eks/add-ons.tf:62]: https://docs.aws.amazon.com/eks/latest/userguide/managing-coredns.html 
v1.10.1-eksbuild.11 per this.

**Comment by zacharyblasczyk** [comment]
1.27:
* aws-efs-csi-driver - `v2.0.4-eksbuild.1`
* aws-ebs-csi-driver - `v1.31.0-eksbuild.1`
* coredns - `v1.10.1-eksbuild.11`
* kube-proxy - `v1.27.12-eksbuild.5`

**Comment by jsbroks** [comment]
This PR is included in version 4.13.0 :tada:

**Pull request merged**